test/Hyper/Trout/RouterSpec.purs

```diff
@@ -119,6 +128,22 @@ spec =
         conn <- makeRequest GET "/wiki/foo/bar/baz.txt"
         testStringBody conn `shouldEqual` "Viewing page: foo&#x2F;bar&#x2F;baz.txt"
 
+      it "matches QueryParam route" do
```

Comment:
What happens if there's no query param `q`, i.e. if the argument to `search` is a `Nothing`? Does argonaut then encode `Nothing` as JSON?